src/flint/fmpq_mpoly.jl

```diff
@@ -108,6 +108,10 @@ end
 
 function coeff(a::QQMPolyRingElem, i::Int)
   z = QQFieldElem()
+  return coeff!(z, a, i)
+end
+
+function coeff!(z::QQFieldElem, a::QQMPolyRingElem, i::Int)
```

Comment:
Dumb question (I could dig through the code but hope you'll just have the answer ready): is this change in here *necessary* for https://github.com/Nemocas/AbstractAlgebra.jl/pull/2196 or "just" an optimization?

Reply:
Ah, I see.

*Not* adding such fallbacks would make it easier for us to find "forgotten" ones. I think I'd kind prefer this. But in either case, `coeff!` should be added to the ring interface documentation to `AA`.

BTW I think these `coeff!` methods are useful on their own, i.e., separately from the benefit of having inplace iteration...

Reply:
I would say that we should have "abstract" fallbacks. In particular, they should be added to https://github.com/Nemocas/AbstractAlgebra.jl/pull/2196 so that the new features works at least works for all flint-backed rings. PRs like the one here can then enable certain optimizations for these types.



> _Not_ adding such fallbacks would make it easier for us to find "forgotten" ones. I think I'd kind prefer this.

We can just have a draft PR later dropping the fallback again, and then we see the forgotton ones from the failures in downstream CI there.